### PR TITLE
[refactor] Replace manual semantic version utilities/functions with semver package

### LINKED ISSUE #4787: Replace all manual semver operations/utils with `semver` package
`semver` package is added as a dependency in https://github.com/Comfy-Org/ComfyUI_frontend/pull/4363. We should replace all manually-implemented semver operations such as all uses of `compareVersion`

https://github.com/Comfy-Org/ComfyUI_frontend/blob/37099c498b6768caa163ae14f524ed1468611d60/src/utils/formatUtil.ts#L406

The `semver` package will generally be more robust and handle more edge-cases and does not require us to maintain coverage of all semver-related functionality.

### PR DESCRIPTION
**Pull request by christian-byrne**
## Summary
- Replace custom `compareVersions()` with `semver.compare()`
- Replace custom `isSemVer()` with `semver.valid()`  
- Remove deprecated version comparison functions from `formatUtil.ts`
- Update all version comparison logic across components and stores
- Fix tests to use semver mocking instead of formatUtil mocking

## Benefits
- **Industry standard**: Uses well-maintained, battle-tested `semver` package
- **Better reliability**: Handles edge cases more robustly than custom implementation
- **Consistent behavior**: All version comparisons now use the same underlying logic
- **Type safety**: Better TypeScript support with proper semver types

Fixes #4787

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5653-refactor-Replace-manual-semantic-version-utilities-functions-with-semver-package-2736d73d365081fb8498ee11cbcc10e2) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/components/dialog/content/manager/PackVersionBadge.vue [hunk 1]
```diff
@@ -43,13 +43,13 @@
 
 <script setup lang="ts">
 import Popover from 'primevue/popover'
+import * as semver from 'semver'
```
Comment (DrJKL) [inline src/components/dialog/content/manager/PackVersionBadge.vue:46]:
```suggestion
import { valid as validSemver } from 'semver'
```

Comment (DrJKL) [reply src/components/dialog/content/manager/PackVersionBadge.vue:46]: Pleeeeease

File: src/components/dialog/content/manager/PackVersionSelectorPopover.vue [hunk 2]
```diff
@@ -142,7 +142,9 @@ onMounted(() => {
     getInitialSelectedVersion() ?? SelectedVersionValues.LATEST
   selectedVersion.value =
     // Use NIGHTLY when version is a Git hash
-    isSemVer(initialVersion) ? initialVersion : SelectedVersionValues.NIGHTLY
+    semver.valid(initialVersion)
```
Comment (DrJKL) [inline src/components/dialog/content/manager/PackVersionSelectorPopover.vue:145]:
```suggestion
    validSemver(initialVersion)
```

File: src/platform/settings/settingStore.ts [hunk 3]
```diff
@@ -132,17 +132,20 @@ export const useSettingStore = defineStore('setting', () => {
...
+          if (!semver.valid(version)) {
             continue
           }
 
-          if (compareVersions(installedVersion, version) >= 0) {
-            const versionedDefault = defaultsByInstallVersion[version]
+          if (semver.compare(installedVersion, version) >= 0) {
+            const versionedDefault =
+              defaultsByInstallVersion[
+                version as keyof typeof defaultsByInstallVersion
```
Comment (DrJKL) [inline src/platform/settings/settingStore.ts:147]: Seems a little dangerous. This would be undefined if it's not actually in the defaults?

File: tests-ui/tests/composables/useUpdateAvailableNodes.test.ts [hunk 4]
```diff
@@ -86,19 +86,19 @@ describe('useUpdateAvailableNodes', () => {
...
-        return !version.includes('nightly')
-      }
-    )
+    mockSemverValid.mockImplementation((version) => {
+      return version &&
+        typeof version === 'string' &&
+        !version.includes('nightly')
+        ? version
+        : null
+    })
```
Comment (DrJKL) [inline tests-ui/tests/composables/useUpdateAvailableNodes.test.ts:95]: You could do this when you setup the mock initially.

File: tests-ui/tests/composables/useUpdateAvailableNodes.test.ts [hunk 5]
```diff
@@ -86,19 +86,19 @@ describe('useUpdateAvailableNodes', () => {
...
-        if (latest === '2.0.0' && installed === '1.0.0') return 1 // outdated
-        if (latest === '1.0.0' && installed === '1.0.0') return 0 // up to date
-        return 0
-      }
-    )
+    mockSemverCompare.mockImplementation((latest, installed) => {
+      if (latest === '2.0.0' && installed === '1.0.0') return 1 // outdated
+      if (latest === '1.0.0' && installed === '1.0.0') return 0 // up to date
+      return 0
+    })
```
Comment (DrJKL) [inline tests-ui/tests/composables/useUpdateAvailableNodes.test.ts:101]: Couldn't you just spy on it and call through?

File: tests-ui/tests/store/releaseStore.test.ts [hunk 6]
```diff
@@ -114,16 +114,16 @@ describe('useReleaseStore', () => {
     })
 
     it('should show update button (shouldShowUpdateButton)', async () => {
-      const { compareVersions } = await import('@/utils/formatUtil')
-      vi.mocked(compareVersions).mockReturnValue(1) // newer version available
+      const semver = await import('semver')
```
Comment (DrJKL) [inline tests-ui/tests/store/releaseStore.test.ts:117]: This can be imported at the top instead of dynamically.

Comment (DrJKL) [reply tests-ui/tests/store/releaseStore.test.ts:117]: Ideally as named imports instead of a namespace import ;-)

**Review by DrJKL — DISMISSED** [review]

File: src/components/dialog/content/manager/PackVersionBadge.vue (hunk 1 above)
Comment (christian-byrne) [reply src/components/dialog/content/manager/PackVersionBadge.vue:46]: Updated to use named import `{ valid as validSemver }` in 99d1b95

File: src/platform/settings/settingStore.ts (hunk 3 above)
Comment (christian-byrne) [reply src/platform/settings/settingStore.ts:147]: Added undefined check to prevent unsafe access in b40d47e

File: tests-ui/tests/store/releaseStore.test.ts (hunk 6 above)
Comment (christian-byrne) [reply tests-ui/tests/store/releaseStore.test.ts:117]: Moved semver imports to top level and used named imports in c568767

File: src/components/dialog/content/manager/PackVersionSelectorPopover.vue (hunk 2 above)
Comment (christian-byrne) [reply src/components/dialog/content/manager/PackVersionSelectorPopover.vue:145]: Updated to use validSemver in 30b8e35

File: tests-ui/tests/composables/useUpdateAvailableNodes.test.ts (hunk 4 above)
Comment (christian-byrne) [reply tests-ui/tests/composables/useUpdateAvailableNodes.test.ts:95]: Good point! The mock setup was already in beforeEach, so this was already addressed.

File: tests-ui/tests/composables/useUpdateAvailableNodes.test.ts (hunk 5 above)
Comment (christian-byrne) [reply tests-ui/tests/composables/useUpdateAvailableNodes.test.ts:101]: You're right\! For this test, mocking with specific test cases was more appropriate than call-through since we need predictable test behavior, but spying with call-through is definitely a good pattern for other scenarios.
